addons/hr/models/hr_employee.py

```diff
@@ -414,11 +414,12 @@ def _get_version(self, date=fields.Date.today()):
         Return the version that should be used for the given date.
         If no valid version is found, we return the very first version of the employee.
         """
-        version = self.env['hr.version'].search([
-            ('employee_id', '=', self.id),
-            ('date_version', '<=', date)],
-            order='date_version desc', limit=1)
-        return version or self.version_ids[0]
+        self.ensure_one()
+        if isinstance(date, str):
+            date = fields.Date.to_date(date)
+        versions = self.version_ids.sorted('date_version')
+        valid_versions = versions.filtered(lambda v: v.date_version <= date)
+        return valid_versions[-1] if valid_versions else versions[:1]
```

Comment:
```suggestion
        return valid_versions[-1] if valid_versions else versions[0]
```
Here, we want to return the recordset not a list of recordsets